Code/Framework/AzCore/AzCore/Asset/AssetSerializer.h

```diff
@@ -120,7 +120,7 @@ namespace AZ {
 
             const Uuid& GetSpecializedTypeId() const override
```

Comment:
question: Should these changes be in a separate PR? Were they somehow contributing to a leak? This is something @AMZN-stankowi should likely take a look at

Reply:
Yeah, you can take a look at the commit message of d10096442ba68bcdecc0fc68ab83954b9303fb0b for the full details. The TL;DR; is the old code stored a map where the keys are typeids. It would use one key to check if the map contained a typeid, and if it didn't, it used a different typeid to store an object in the map. The key used to store the item was the same key for many different lookups, so the value stored at that key would get overwritten. Because the value type is a raw pointer, the overwritten pointer would cause a leak.

This fixes the leak by making the key used for lookup and the key used for insertion the same.

As for whether this should be in a separate PR, I don't think it matters. This change was done in its own commit, and I plan to merge it with a merge commit (not squash), so if this one commit causes issues down the line, the commit can be reverted independently of the rest of the changes from this PR.

Reply:
Aaah I see interesting! Thank you for the explanation, much appreciated. Sure thing makes sense :+1: I just wasn't sure on the surface if it was related, but it clearly is! 😅 Thanks!